stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/attachments/content/GiphyAttachmentContent.kt

```diff
@@ -70,10 +87,47 @@ public fun GiphyAttachmentContent(
         "Missing preview URL."
     }
 
+    val density = LocalDensity.current
+
     val painter = rememberStreamImagePainter(attachment.imagePreviewUrl)
 
+    val giphyInfo = attachment.giphyInfo(giphyInfoType)
+
+    val maxWidth = ChatTheme.dimens.attachmentsContentGiphyMaxWidth
+    val maxHeight = ChatTheme.dimens.attachmentsContentGiphyMaxHeight
+
+    val giphyDimensions: DpSize by remember(key1 = giphyInfo, key2 = upscaleFactor) {
+        derivedStateOf {
+            if (giphyInfo != null) {
+                with(density) {
+                    val giphyWidth = (giphyInfo.width * abs(upscaleFactor)).toDp()
+                    val giphyHeight = (giphyInfo.height * abs(upscaleFactor)).toDp()
+
+                    when {
+                        upscaleFactor == StreamAttachmentFactories.GIPHY_FILL_MAX_SPACE -> calculateResultingDimensions(
```

Comment:
If we decide that we only want to fill max space because pixel faithful renditions are too small then this can be quite simplified.